lib/alexa.js

```diff
@@ -1,418 +1,573 @@
+ *
+ * @fires eventString based off event.request
+ * @return {undefined}
+ */
 function EmitEvent() {
-    this.state = this._event.session.attributes[_StateString] || '';
-
-    var eventString = '';
-
-    if (this._event.session['new'] && this.listenerCount('NewSession' + this.state) === 1) {
-        eventString = 'NewSession';
-    } else if(this._event.request.type === 'LaunchRequest') {
-        eventString = 'LaunchRequest';
-    } else if(this._event.request.type === 'IntentRequest') {
-        eventString = this._event.request.intent.name;
-    } else if (this._event.request.type === 'SessionEndedRequest'){
-        eventString = 'SessionEndedRequest';
-    } else if (this._event.request.type.substring(0,11) === 'AudioPlayer') {
-        eventString = this._event.request.type.substring(12);
-    } else if (this._event.request.type.substring(0,18) === 'PlaybackController') {
-        eventString = this._event.request.type.substring(19);
-    }
-
-    eventString += this.state;
-
-    if(this.listenerCount(eventString) < 1) {
-        eventString = 'Unhandled' + this.state;
-    }
-
-    if(this.listenerCount(eventString) < 1){
-        throw new Error(`No 'Unhandled' function defined for event: ${eventString}`);
-    }
-
-    this.emit(eventString);
+  this.state = this._event.session.attributes[_StateString] || '';
+
+  let eventString = '';
+
+  if (this._event.session.new && this.listenerCount(`NewSession${this.state}`) === 1) {
+    eventString = 'NewSession';
+  } else if (this._event.request.type === 'LaunchRequest') {
+    eventString = 'LaunchRequest';
+  } else if (this._event.request.type === 'IntentRequest') {
+    eventString = this._event.request.intent.name;
+  } else if (this._event.request.type === 'SessionEndedRequest') {
+    eventString = 'SessionEndedRequest';
+  } else if (this._event.request.type.substring(0, 11) === 'AudioPlayer') {
+    eventString = this._event.request.type.substring(12);
+  } else if (this._event.request.type.substring(0, 18) === 'PlaybackController') {
+    eventString = this._event.request.type.substring(19);
+  }
+
+  eventString += this.state;
+
+  if (this.listenerCount(eventString) < 1) {
+    eventString = `Unhandled${this.state}`;
+  }
+
+  if (this.listenerCount(eventString) < 1) {
+    throw new Error(`No 'Unhandled' function defined for event: ${eventString}`);
+  }
+
+  this.emit(eventString);
 }
 
-function RegisterHandlers() {
-    for(var arg = 0; arg < arguments.length; arg++) {
-        var handlerObject = arguments[arg];
-
-        if(!isObject(handlerObject)) {
-            throw new Error(`Argument #${arg} was not an Object`);
-        }
+/**
+ * RegisterHandlers - Add all handlers to this, adding events for each property.
+ *
+ * NB: this keyword must be bound to handler object.
+ *
+ * @param  {...Object} args  Array of all handlers to register
+ * @return {undefined}
+ */
+function RegisterHandlers(...args) {
+  for (let arg = 0; arg < args.length; arg += 1) {
```

Comment:
I don't think it's that important that I like the implementation as I'm not the owner of this repo nor am I in any way related to amazon 😄 
It just would fit into this merge request as it is a ES6 change.